datalad_debian/bootstrap_builder.py

```diff
@@ -58,22 +65,40 @@ class BootstrapBuilder(Interface):
             doc="""specify a builder dataset that contains a build environment
             configuration""",
             constraints=EnsureDataset() | EnsureNone()),
+        template=Parameter(
```

Comment:
Maybe `flavor` is a more fitting name?

Or, as a different angle: With could turn `cfgtype` into a proper mode switch to select the to-be-employed technology, but no longer have any implications on the "template" filename (for the recipe at least).

WDYT?

Reply:
> Or, as a different angle: With could turn `cfgtype` into a proper mode switch to select the to-be-employed technology, but no longer have any implications on the "template" filename (for the recipe at least).

I think it sounds nice, but how would it work? Do I understand correctly that cfgtype wouldn't be part of the recipe names, and the mode switch would only influence the build cmd?

Reply:
Yes, something like that. One option to label the environment, and one option to select the right builder for that environment.

That being said, I have likely not thought this through as much as you did.

Reply:
Hold on. Now I got it. `template` refers to the instantiated "template". This is what threw me off. What about this for a terminology:

- template: this is what `configure-builder` takes as input
- recipe: this is what `configure-builder`'s output is called (the instantiated template with a placeholder fulfilled)

So when `bootstrap-builder` needs to run, we can give it a recipe filename and either

- decode the handler (e.g. singularity), and flavor from the filename by enshrining a syntax for it
- take the recipe name verbatim/uninterpreted and require specification of the handler

Both approaches would be fine from my POV